packages/react/src/hooks/useVisibilityLoader.js

```diff
@@ -0,0 +1,75 @@
+import { useEffect, useLayoutEffect, useState } from 'react';
+import warning from 'warning';
+
+import { browserSupports } from '../utils/componentUtilityFunctions';
+
+const useVisibilityLoader = (ref, { maxToLoad = 1, loadBatchSize = 20 } = {}) => {
+  if (!browserSupports('IntersectionObserver')) {
+    warning(
+      !__DEV__,
+      'The current browser does not support ResizeObserver. You will need to include a ResizeObserver polyfill for this component to function properly.'
```

Comment:
need to update the message to reflect particular observer or generalize if its all or nothing support wise.

Reply:
updated.